samcli/commands/delete/delete_context.py

```diff
@@ -0,0 +1,163 @@
+"""
+Delete a SAM stack
+"""
+
+import boto3
+
+# import docker
+import click
+from click import confirm
+from click import prompt
+from samcli.cli.cli_config_file import TomlProvider
+from samcli.lib.utils.botoconfig import get_boto_config_with_user_agent
+from samcli.lib.delete.cf_utils import CfUtils
+from samcli.lib.package.s3_uploader import S3Uploader
+from samcli.lib.package.artifact_exporter import mktempfile, get_cf_template_name
+
+# from samcli.yamlhelper import yaml_parse
+
+# Intentionally commented
+# from samcli.lib.package.artifact_exporter import Template
+# from samcli.lib.package.ecr_uploader import ECRUploader
+# from samcli.lib.package.uploaders import Uploaders
+
+CONFIG_COMMAND = "deploy"
+CONFIG_SECTION = "parameters"
+TEMPLATE_STAGE = "Original"
+
+
+class DeleteContext:
+    def __init__(self, stack_name, region, profile, config_file, config_env):
+        self.stack_name = stack_name
+        self.region = region
+        self.profile = profile
+        self.config_file = config_file
+        self.config_env = config_env
+        self.s3_bucket = None  # s3_bucket
+        self.s3_prefix = None  # s3_prefix
+        self.cf_utils = None
+        self.start_bold = "\033[1m"
+        self.end_bold = "\033[0m"
```

Comment:
ANSI escape characters aren't compatible on Windows AFAIK; we should be using higher-level libraries. I think `click` supports this.